File: trunk/CharasLike/Windows/EasyPosition.vb
Public Class EasyPosition

    Private Sub Form5_Deactivate(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Deactivate
        If Me.Visible() Then
            Me.Close()
        End If
    End Sub

    Private Sub Form5_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If LayersWindow.CurrentNode Is Nothing Then
            Label1.Text = ""
            Return
        End If

        Label1.SendToBack()
        Label1.Text = LayersWindow.CurrentNode.Layer.Offset.X.ToString + "," + LayersWindow.CurrentNode.Layer.Offset.Y.ToString()
    End Sub

    Private Sub Form5_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed
    End Sub

    Private Sub Button9_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button9.Click
        If LayersWindow.CurrentNode Is Nothing Then Return
        LayersWindow.CurrentNode.Layer.Offset = New Point(0, 0)
        Canvas.UpdateDrawing()
        Label1.Text = LayersWindow.CurrentNode.Layer.Offset.X.ToString + "," + LayersWindow.CurrentNode.Layer.Offset.Y.ToString()
    End Sub

    Public Sub OffsetIt(ByVal Pnt As Point)
        If LayersWindow.CurrentNode Is Nothing Then Return
        LayersWindow.CurrentNode.Layer.Offset += Pnt
        Canvas.UpdateDrawing()
        Label1.Text = LayersWindow.CurrentNode.Layer.Offset.X.ToString + "," + LayersWindow.CurrentNode.Layer.Offset.Y.ToString()
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        OffsetIt(New Point(0, -1))
    End Sub

    Private Sub Button8_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button8.Click
        OffsetIt(New Point(1, -1))
    End Sub

    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        OffsetIt(New Point(1, 0))
    End Sub

    Private Sub Button7_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button7.Click
        OffsetIt(New Point(1, 1))
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        OffsetIt(New Point(0, 1))
    End Sub

    Private Sub Button6_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button6.Click
        OffsetIt(New Point(-1, 1))
    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        OffsetIt(New Point(-1, 0))
    End Sub

    Private Sub Button5_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button5.Click
        OffsetIt(New Point(-1, -1))
    End Sub
End Class

File: Windows/EasyPosition.cs
using System.Collections.Generic;
using System;
using System.Drawing;
using System.Diagnostics;
using System.Collections;
using System.Windows.Forms;

namespace OpenCharas
{
	public partial class EasyPosition
	{
		public EasyPosition()
		{
			InitializeComponent();
		}

		public void Form5_Deactivate(System.Object sender, System.EventArgs e)
		{
			if (this.Visible)
				this.Close();
		}

		public void Form5_Load(System.Object sender, System.EventArgs e)
		{
			if (Program.layersWindowForm.CurrentNode == null)
			{
				Label1.Text = "";
				return;
			}

			Label1.SendToBack();
			Label1.Text = Program.layersWindowForm.CurrentNode.Layer.Offset.X.ToString() + "," + Program.layersWindowForm.CurrentNode.Layer.Offset.Y.ToString();
		}

		public void Form5_FormClosed(System.Object sender, System.Windows.Forms.FormClosedEventArgs e)
		{
		}

		public void Button9_Click(System.Object sender, System.EventArgs e)
		{
			if (Program.layersWindowForm.CurrentNode == null)
				return;

			Program.layersWindowForm.CurrentNode.Layer.Offset = new Point(0, 0);
			Program.canvasForm.UpdateDrawing();
			Label1.Text = Program.layersWindowForm.CurrentNode.Layer.Offset.X.ToString() + "," + Program.layersWindowForm.CurrentNode.Layer.Offset.Y.ToString();
		}

		public void OffsetIt(Point Pnt)
		{
			if (Program.layersWindowForm.CurrentNode == null)
				return;

			Program.layersWindowForm.CurrentNode.Layer.Offset = new Point(Program.layersWindowForm.CurrentNode.Layer.Offset.X + Pnt.X, Program.layersWindowForm.CurrentNode.Layer.Offset.Y + Pnt.Y);
			Program.canvasForm.UpdateDrawing();
			Label1.Text = Program.layersWindowForm.CurrentNode.Layer.Offset.X.ToString() + "," + Program.layersWindowForm.CurrentNode.Layer.Offset.Y.ToString();
		}

		public void Button1_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(0, -1));
		}

		public void Button8_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(1, -1));
		}

		public void Button4_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(1, 0));
		}

		public void Button7_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(1, 1));
		}

		public void Button2_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(0, 1));
		}

		public void Button6_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(-1, 1));
		}

		public void Button3_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(-1, 0));
		}

		public void Button5_Click(System.Object sender, System.EventArgs e)
		{
			OffsetIt(new Point(-1, -1));
		}
	}
}
